### PR TITLE
Fixed following CRs:

### PR DESCRIPTION
**Pull request by saifuddin-xilinx**
CR-1054109
    CR-1054175

Simplefied the rescan logic. 
Removed the rescan logic to rescan  from individual RootPort 
Now we are rescanning from the /sys/bus/pci/rescan.

### CODE REVIEW

**Comment by maxzhen** [comment]
This code may have some issues. I think that you want to use Lizhi's code in SC update code path where he also remove and rescan for a user function. Anyway, I'll let Lizhi chime in here. Just added him as a reviewer.

**Comment by saifuddin-xilinx** [comment]
> Please work with @houlz0507 for the correct approach that he has implemented in SC upgrade code.

I am currently working on it. Will update once I am done.

File: src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp [hunk 1]
```diff
@@ -187,26 +188,58 @@ static int removeDevice(unsigned int index, bool is_userpf)
...
 
+    /* Poll till remove is done */
+    do {
+        sleep(1);
+        std::string path = dev->get_sysfs_path("", "remove");
+        if (!boost::filesystem::exists(path))
+            break;
+
+    } while (++wait < POLL_TIMEOUT );
+
```
Comment (houlz0507) [inline src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp]: Why polling remove? Is echo 1 >/sys/.../remove a sync call?

File: src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp [hunk 2]
```diff
@@ -187,26 +188,58 @@ static int removeDevice(unsigned int index, bool is_userpf)
...
+        if (iter->path().filename() == "mgmt_pf")
+            return true;
+
+        ++iter;
+    }
+
+    return false;
+}
+
+
```
Comment (houlz0507) [inline src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp]: why not using "pcidev::get_dev(index, false);" to get mgmtpf?

File: src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp [hunk 3]
```diff
@@ -226,59 +259,60 @@ static boost::filesystem::path findXilinxRootPort(void)
...
+            ofile << 1;
+            ofile.flush();
+            if (!ofile.good()) 
+            {
+                std::cout << "Failed to write " << iPath << ":"  << strerror(errno) << std::endl;
+                ofile.close();
+                return -errno;
+            }
+        }
```
Comment (houlz0507) [inline src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp]: "mgmt_dev->sysfs_put("", "dparent/rescan", errmsg, "1\n");" will rescan specific card.

File: src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp (hunk 1 above)
Comment (saifuddin-xilinx) [reply src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp]: In some setup ever after echo 1 > remove comeout successfully,, the sysfs files are taking some time to disappear. and xbmgmt scan is still showing the all the xilinx devices. Hence some of the test scripts are failing. Please check the below output:

[root@xsjdsv08 xbmgmt]# ./xbmgmt  scan
 0000:89:00.0 xilinx_samsung_u2x4_201920_2(ID=0x5e3d30a2) mgmt(inst=35072)
 0000:3e:00.0 xilinx_samsung_u2x4_201920_2(ID=0x5e3d30a2) mgmt(inst=15872)
[root@xsjdsv08 xbmgmt]# ls /sys/bus/pci/devices/0000\:89\:00.0/mgmt_pf 
/sys/bus/pci/devices/0000:89:00.0/mgmt_pf
[root@xsjdsv08 xbmgmt]# ./xbmgmt hotplug --offline 0000:89:00.0
CAUTION: Performing hotplug command. This command is going to impact both user pf and mgmt pf.
Please make sure no application is currently running.
Are you sure you wish to proceed? [y/n]: y
[root@xsjdsv08 xbmgmt]# ls /sys/bus/pci/devices/0000\:89\:00.0/mgmt_pf 
/sys/bus/pci/devices/0000:89:00.0/mgmt_pf
[root@xsjdsv08 xbmgmt]# ./xbmgmt  scan
 0000:89:00.0 xilinx_samsung_u2x4_201920_2(ID=0x5e3d30a2) mgmt(inst=35072)
 0000:3e:00.0 xilinx_samsung_u2x4_201920_2(ID=0x5e3d30a2) mgmt(inst=15872)
[root@xsjdsv08 xbmgmt]# ls /sys/bus/pci/devices/0000\:89\:00.0/mgmt_pf 
ls: cannot access /sys/bus/pci/devices/0000:89:00.0/mgmt_pf: No such file or directory
[root@xsjdsv08 xbmgmt]#

Comment (houlz0507) [reply src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp]: In the hotplug case, please echo 1 > <parent>/remove. That should be a sync call and you should not see both userpf and mgmtpf node after echo 1 returned.

**Comment by saifuddin-xilinx** [comment]
Created an another pull request for the same.

**Pull request closed without merging**